src/poketch_data.c

```diff
@@ -0,0 +1,360 @@
+    s8 appIndex;                                //!< Currently selected app
+    u8 appRegistry[POKETCH_REGISTRY_SIZE];      //!< Registration status of all apps. Indices 0-24 correspond to the App IDs in poketch_data.h. Indices 25-31 are unused.
+    
+    u32 pedometer;         //!< Step counter
+
+    u16 alarmSet : 1;      //!< Whether or not the alarm is currently enabled.
+    u16 alarmHour : 5;     //!< Current Hour setting on the alarm clock.
+    u16 alarmMinute : 6;   //!< Current Minute setting on the alarm clock.
+    
+    u16 unk_28_12 : 4;     //!< unused
+
+    u8 unk_2A[120];
+    u32 unk_A4;
+    u8 unk_A8;
+
+    struct {
+        u8 x;                                       //!< X coordinate of map marker
+        u8 y;                                       //!< Y coordinate of map marker
+    } markMapPositions[POKETCH_MAPMARKER_COUNT];    //!< Map markers
+
+    struct {
+        u16 unk_00;
+        u16 unk_02;
+        u32 unk_04;
+    } unk_B8[12];
+} PoketchData;
+
+
+int Poketch_SaveSize (void)
+{
+    return sizeof(PoketchData);
+}
+
+void Poketch_Init (PoketchData * poketchData)
+{
+    int i;
+
+    for (i = 0; i < POKETCH_REGISTRY_SIZE; i++) {
+        poketchData->appRegistry[i] = 0;
+    }
+
+    poketchData->appCount = 0;
+    poketchData->appIndex = 0;
+    poketchData->unk_00_0 = 0;
+    poketchData->screenColor = 0;
+    poketchData->pedometerEnabled = 0;
+    poketchData->pedometer = 0;
+    poketchData->alarmSet = 0;
+    poketchData->alarmHour = 0;
+    poketchData->alarmMinute = 0;
+    poketchData->unk_A8 = 1;
+    poketchData->unk_A4 = 0;
+
+    {
+        static const struct {
+            u8 x;
+            u8 y;
+        } v1[POKETCH_MAPMARKER_COUNT] = {
+            {104, 152},
+            {120, 152},
+            {136, 152},
+            {152, 152},
+            {168, 152},
+            {184, 152}
+        };
+
+        for (i = 0; i < POKETCH_MAPMARKER_COUNT; i++) {
+            poketchData->markMapPositions[i].x = v1[i].x;
+            poketchData->markMapPositions[i].y = v1[i].y;
+        }
+    }
+
+    for (i = 0; i < 12; i++) {
+        poketchData->unk_B8[i].unk_00 = 0;
+        poketchData->unk_B8[i].unk_02 = 0;
+        poketchData->unk_B8[i].unk_04 = 0;
+    }
+
+    poketchData->unk_00_2 = 0;
+    PoketchData_RegisterApp(poketchData, POKETCH_APPID_DIGITALWATCH);
+}
+
+void sub_020567D0 (PoketchData * poketchData)
+{
+    poketchData->unk_00_0 = 1;
+}
+
+BOOL sub_020567E0 (PoketchData * poketchData)
+{
+    return poketchData->unk_00_0;
+}
+
+BOOL PoketchData_CheckAppRegistered (PoketchData * poketchData, int appID)
+{
+    return poketchData->appRegistry[appID];
+}
+
+BOOL PoketchData_RegisterApp (PoketchData * poketchData, int appID)
+{
+    BOOL appRegistered = 0;
+
+    GF_ASSERT(appID >= 0 && appID < POKETCH_APPID_MAX);
+
+    if (poketchData->appCount < POKETCH_APPID_MAX) {
+        if (poketchData->appRegistry[appID] == 0) {
```

Comment:
Prefer logical conjunction over nesting ifs, as it reduces the overall nesting of the code.

```suggestion
    if (poketchData->appCount < POKETCH_APPID_MAX
            && poketchData->appRegistry[appID] == 0) {
```